src/RoadRegistry.Syndication.ProjectionHost/Mapping/EventSerializerMapping.cs

```diff
@@ -0,0 +1,42 @@
+namespace RoadRegistry.Syndication.ProjectionHost.Mapping
+{
+    using System.Collections.Generic;
+    using System.Collections.ObjectModel;
+    using System.Linq;
+    using System.Reflection;
+    using System.Runtime.Serialization;
+    using Projections.MunicipalityEvents;
+
+    public class EventSerializerMapping
+    {
+        private readonly IReadOnlyDictionary<string, DataContractSerializer> _eventSerializers;
+
+        public EventSerializerMapping()
+        {
+            _eventSerializers = new ReadOnlyDictionary<string, DataContractSerializer>(
+                Assembly
+                    .GetAssembly(typeof(MunicipalityWasRegistered))
+                    ?.GetTypes()
+                    .Where(t => t.Namespace != null &&
+                                (t.Namespace.EndsWith("MunicipalityEvents") ||
+                                 t.Namespace.EndsWith("StreetNameEvents")))
+                    .ToDictionary(
+                        type => type.Name,
+                        type => new DataContractSerializer(type)));
+        }
+
+        public DataContractSerializer GetSerializerFor(string eventName)
+        {
+            if (_eventSerializers.ContainsKey(eventName))
+                return _eventSerializers[eventName];
+
+            throw new KeyNotFoundException($"Serializer voor event met naam '{eventName}' niet gevonden.");
```

Comment:
This repository uses English, please stick to it. If you find more places in the codebase that violate this, feel free to point it out and / or fix it.